Formula/g/glib.rb

```diff
@@ -36,10 +35,17 @@ class Glib < Formula
     depends_on "util-linux"
   end
 
-  # These used to live in the now defunct `glib-utils`.
   link_overwrite "bin/gdbus-codegen", "bin/glib-genmarshal", "bin/glib-mkenums", "bin/gtester-report"
   link_overwrite "share/glib-2.0/codegen", "share/glib-2.0/gdb"
 
+  # Fix python 3.12 compatibility, remove in 2.80
+  patch do
+    url "https://gitlab.gnome.org/GNOME/glib/-/commit/6ef967a0f930ce37a8c9b5aff96969.diff"
```

Comment:
This uses `packaging` from PyPI rather than things from the standard library, so we might need to change how we package this to vendor that somewhere.

In any case, we should make sure we have a test that covers this.